datafusion/src/physical_plan/file_format/parquet.rs

```diff
@@ -204,11 +178,8 @@ impl ParquetExec {
     }
 
     /// List of data files
-    pub fn partitions(&self) -> Vec<&[PartitionedFile]> {
-        self.partitions
-            .iter()
-            .map(|fp| fp.file_partition.files.as_slice())
-            .collect()
+    pub fn file_groups(&self) -> &[Vec<PartitionedFile>] {
```

Comment:
We replace `partitions` with `file_groups` to decrease the overuse of the term "partition" which represents different things in different contexts:
- on the listing table side, a partition refer to a "hive partition", that is to say a set of files grouped into a folder because they share a common attribute
- on the execution plan side, a partition is a unit of parallelism. Files are grouped together to provide a good workload for one thread/executor.